test/Interop/Cxx/constructor/copy-basic.swift

```diff
@@ -0,0 +1,55 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXCopyConstructorTestSuite = TestSuite("CXX Copy Constructor")
+
+@inline(never)
+func createHasUserProvidedCopyConstructor() -> HasUserProvidedCopyConstructor {
+  let box = HasUserProvidedCopyConstructor(42)
+  return box
+}
+
+@inline(never)
+func createTypeWithEmptyCopyConstructor() -> EmptyCopyConstructor {
+  let empty = EmptyCopyConstructor()
+  return empty
+}
+
+@inline(never)
+func createTypeWithNonTrivialImplicitCopyConstructor() -> HasNonTrivialImplicitCopyConstructor {
+  let implicit = HasNonTrivialImplicitCopyConstructor()
+  return implicit
+}
+
+@inline(never)
+func createTypeWithNonTrivialDefaultCopyConstructor() -> HasNonTrivialDefaultCopyConstructor {
+  let def = HasNonTrivialDefaultCopyConstructor()
+  return def
+}
+
+// It doesn't really matter how many copies were made (as long as it's at least
+// one). What we really want to be checking is that the correct copy constructor
+// was eventually called. So check that the current value is greater than the
+// initial value.
```

Comment:
As noted above, I would suggest using an explicit `numCopies` member variable to make these tests more self-explanatory.